src/main/java/io/fabric8/maven/docker/util/DockerFileUtil.java

```diff
@@ -55,23 +57,46 @@ private DockerFileUtil() {}
      */
     public static List<String> extractBaseImages(File dockerFile, FixedStringSearchInterpolator interpolator) throws IOException {
         List<String[]> fromLines = extractLines(dockerFile, "FROM", interpolator);
+        Map<String, String> args = extractArgs(dockerFile, interpolator);
         Set<String> result = new LinkedHashSet<>();
         Set<String> fromAlias = new HashSet<>();
         for (String[] fromLine :  fromLines) {
             if (fromLine.length > 1) {
                 if (fromLine.length == 2) { // FROM image:tag use case
-                    result.add(fromLine[1]);
+                    result.add(resolveImageTagFromArgs(fromLine[1], args));
                 } else if (fromLine.length == 4) { // FROM image:tag AS alias use case
                     if (!fromAlias.contains(fromLine[1])) {
-                        result.add(fromLine[1]);
+                        result.add(resolveImageTagFromArgs(fromLine[1], args));
                     }
-                    fromAlias.add(fromLine[3]);
+                    fromAlias.add(resolveImageTagFromArgs(fromLine[3], args));
                 }
             }
         }
         return result.stream().collect(Collectors.toList());
     }
 
+    /**
+     * Extract Args from dockerfile. All lines containing ARG is taken.
+     *
+     * @param dockerFile file from where to extract args
+     * @param interpolator interpolator for replacing properties
+     * @return HashMap of arguments or empty collection if none is found
+     * @throws IOException in case of any problem reading file
+     */
+    public static Map<String, String> extractArgs(File dockerFile, FixedStringSearchInterpolator interpolator) throws IOException {
+        List<String[]> argLines = extractLines(dockerFile, "ARG", interpolator);
+        Map<String, String> result = new HashMap<>();
+        for (String[] argLine : argLines) {
+            if (argLine.length > 1) {
+                AbstractMap.SimpleEntry<String, String> resolvedArg = resolveArgsFromArgString(argLine[1]);
```

Comment:
I would call the method `updateMapWithArgValue()` and feed in the map as argument. That allows you to avoid to generate temporary, short-lived obkects like `resolvedArg`